cmd/platform/main.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+func findMattermostBinary() string {
+	for _, file := range []string{"./mattermost", "../mattermost", "./bin/mattermost"} {
+		path, _ := filepath.Abs(file)
+		if stat, err := os.Stat(path); err == nil && !stat.IsDir() {
+			return path
+		}
+	}
+	return "./mattermost"
+}
+
+func main() {
+	// Print angry message to use mattermost command directly
+	fmt.Println(`
+------------------------------------ ERROR ------------------------------------------------
+The platform binary has been depricated, please switch to using the new mattermost binary.
+The platform binary will be removed in a future version.
+-------------------------------------------------------------------------------------------
+	`)
+
+	// Execve the real MM binary
+	args := os.Args
+	args[0] = "mattermost"
+	args = append(args, "--platform")
+	if err := syscall.Exec(findMattermostBinary(), args, nil); err != nil {
```

Comment:
For me, looks like a bit error prone. I would use the exact same executable, and print the warning if the excutable name (os.Args[0]) is "platform".

Reply:
Actually, I'm thinking about it and can complain about any binary that isn't call "mattermost".

Reply:
Not sure what could go wrong here that you are worried about. If I just call it directly we get another large binary, I would rather avoid that with this.

Reply:
Basically, how you find the mattermost binary, looks like something that can fail for a lot of reasons, for example, when i build with `make build-linux` it is nstalled in my `$GOCODE/bin/platform` and my repo is in `$GOCODE/src/github.com/mattermost/mattermost-server` if I execute while I in my mattermost-server I wont be able to find  properly the mattermost binary. Probably, with other environment configuration, or system wide installation, or how distribution package our app, it can fail in a lot of different ways finding the mattermost binary. If we want to keep this approach I would prefer to use something like "Detect the current path of the executed file, and search for mattermost binary in the same directory". (Maybe using this https://tip.golang.org/pkg/os/#Executable)

Reply:
Actually checking the disclaimer on that about symlinks and stuff I am inclined to not do it that way and do it this way as we know it works.

Reply:
Not in my machine.
In a clean mattermost-server directory (without compiled files), if I execute make build-linux and the platform, this is the result:

```

------------------------------------ ERROR ------------------------------------------------
The platform binary has been deprecated, please switch to using the new mattermost binary.
The platform binary will be removed in a future version.
-------------------------------------------------------------------------------------------
	
Could not start Mattermost, use the mattermost command directly.
```

Reply:
But this is for compatibility with installs, not dev environments. I you have MM installed it will always be in a place where this would work, it has to be because we find our static files in the same way.

Reply:
Ok, I understand that, anyway I still think we can support dev environment, other installations (for example people who download, compile and install manually the things), other non-official installations like distributions which can be use any other files structure, and would be compatible with the embeded statics approach, which will allow you to execute anywhere.

Anyway, if you have a strong opinion about this I'm ok to merge as is now.

Reply:
All of that is great improvements that we can make to the Mattermost binary. The platform one is just temporary for compatibility on standard installs. Will merge.